### PR TITLE
Issue #147 r spatial

### PR DESCRIPTION
**Pull request by sokole**
Update to plot_sites to fix issue #147

### CODE REVIEW

File: R/plot_taxa.R [hunk 1]
```diff
@@ -1236,51 +1236,72 @@ plot_taxa_abund <- function(data,
...
+#'plot_sites(stacked_data)
+#'
+#'
 #' # Plot the example dataset
 #' plot_sites(ants_L1)
-#' 
-plot_sites <- function(data,
-                       id = NA_character_,
-                       alpha = 1,
-                       labels = TRUE) {
```
Comment (clnsmth) [inline R/plot_taxa.R:1272]: @sokole @RafaelRangel0 Removing the `labels` parameter here breaks backwards compatibility. Any changes to backwards compatibility should be accompanied by deprecation warnings and the behavior phased out over the course of several months to allow users time to modify their applications.

For an example of the preferred deprecation approach, please see: https://github.com/EDIorg/ecocomDP/blob/6b56fa722af128e1702ead4dde79caa5b5d73a50/R/plot_taxa.R#L113

File: R/plot_taxa.R [hunk 2]
```diff
@@ -1236,51 +1236,72 @@ plot_taxa_abund <- function(data,
 
 
 
-#' Plot sites on US map
+#' Work-in-progress upgraded plot_sites
```
Comment (clnsmth) [inline R/plot_taxa.R]: @sokole @RafaelRangel0 is this title change intended? It looks like it may have been a placeholder while development was in progress.

File: R/plot_taxa.R [hunk 3]
```diff
@@ -1236,51 +1236,72 @@ plot_taxa_abund <- function(data,
...
 #'
 #' @param data (list or tbl_df, tbl, data.frame) The dataset object returned by \code{read_data()}, a named list of tables containing the observation and taxon tables, or a flat table containing columns of the observation and location tables.
 #' @param id (character) Identifier of dataset to be used in plot subtitles. Is automatically assigned when \code{data} is a dataset object containing the \code{id} field, or is a table containing the package_id column.
 #' @param alpha (numeric) Alpha-transparency scale of data points. Useful when many data points overlap. Allowed values are between 0 and 1, where 1 is 100\% opaque. Default is 1.
-#' @param labels (logical) Argument to show labels of each US state. Default is TRUE.
```
Comment (clnsmth) [inline R/plot_taxa.R:1244]: @sokole @RafaelRangel0 removing this parameter creates a backwards compatibility breaking change. See below.

File: R/plot_taxa.R [hunk 4]
```diff
@@ -1236,51 +1236,72 @@ plot_taxa_abund <- function(data,
...
-#' 
+#' @param color_var (character) Argument to assign colors to the points on the plot
```
Comment (clnsmth) [inline R/plot_taxa.R:1240]: @sokole @RafaelRangel0 I'm wondering if names like `point_color` and `point_shape` could be more informative.

File: R/plot_taxa.R [hunk 5]
```diff
@@ -1309,23 +1332,33 @@ plot_sites <- function(data,
 
   cleaned <- flat_data %>%
     dplyr::select(
-      .data$longitude,
```
Comment (clnsmth) [inline R/plot_taxa.R:1312]: @sokole @RafaelRangel0 This line of code was changed from `.data$longitude` to `"longitude"`. If I recall correctly, the original code is required for building the package successfully. What is the reason for the change here?

File: R/plot_taxa.R [hunk 6]
```diff
@@ -1296,7 +1296,7 @@ plot_sites <- function(
   validate_arguments(fun.name = "plot", fun.args = as.list(environment()))
 
   # check for suggested packages that are required for this function to work
-  suggs <- c("ggrepel", "usmap","maps")
+  suggs <- c("ggrepel", "usmap","maps","sf")
```
Comment (clnsmth) [inline R/plot_taxa.R]: @sokole @RafaelRangel0 The "maps" package is not listed in the package DESCRIPTION file. Should it be?

**Review by clnsmth — COMMENTED** [review]
@sokole @RafaelRangel0 A few things to resolve before we can merge. Happy to discuss.

File: R/plot_taxa.R (hunk 4 above)
Comment (sokole) [reply R/plot_taxa.R:1240]: sure, we were trying to conform to the existing naming conventions for other plotting functions in the package, but others do not have these sorts of arguments.

File: R/plot_taxa.R (hunk 3 above)
Comment (sokole) [reply R/plot_taxa.R:1244]: We can add it back in if desired.

File: R/plot_taxa.R (hunk 2 above)
Comment (sokole) [reply R/plot_taxa.R]: this is not in the pull request. Looks like you were looking at an older commit that I reverted.

File: R/plot_taxa.R (hunk 6 above)
Comment (sokole) [reply R/plot_taxa.R]: I can add this in

File: R/plot_taxa.R (hunk 5 above)
Comment (sokole) [reply R/plot_taxa.R:1312]: @clnsmth upon closer investigation, I don't think we need the `.data$` if the column names are quoted. This might be updated syntax with recent updates to tidyverse

File: R/plot_taxa.R (hunk 3 above)
Comment (sokole) [reply R/plot_taxa.R:1244]: added back, but `labels` doesn't appear to have actually done anything in the old code.

File: R/plot_taxa.R (hunk 5 above)
Comment (sokole) [reply R/plot_taxa.R:1312]: ah, but these columns were unquoted in the plotting call. I fixed that below.
